keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -571,10 +572,11 @@ class SqlDeliveryConfigRepository(
           ENVIRONMENT_ARTIFACT_CONSTRAINT.TYPE.eq(type)
         )
         .let {
-          // For backwards-compatibility, only use the artifact reference in the query when passed in
+          // For backwards-compatibility, only use the artifact reference in the query when passed in, but also match
+          // rows with a missing reference.
           if (artifactReference != null) {
-            it.and(
-              ENVIRONMENT_ARTIFACT_CONSTRAINT.ARTIFACT_REFERENCE.eq(artifactReference)
+            it.and(ENVIRONMENT_ARTIFACT_CONSTRAINT.ARTIFACT_REFERENCE.eq(artifactReference)
+              .or(ENVIRONMENT_ARTIFACT_CONSTRAINT.ARTIFACT_REFERENCE.isNull)
```

Comment:
This should allow updating those rows in the DB which currently have a `NULL` in the column.